src/iarc7_motion/iarc_tasks/task_utilities/height_holder.py

```diff
@@ -18,20 +19,46 @@ def __init__(self, desired_height = None):
             self._MIN_MANEUVER_HEIGHT = rospy.get_param('~min_maneuver_height')
             self._MAX_Z_ERROR = rospy.get_param('~max_z_error')
             self._K_Z = rospy.get_param('~p_term_height_hold_z')
+            self._DEADZONE = rospy.get_param('~deadzone_height_hold_z')
+            self._DEADZONE_HYSTERESIS = rospy.get_param('~deadzone_hysteresis_height_hold_z')
+            self._DEBUG = rospy.get_param('~debug_height_hold_z')
         except KeyError as e:
             rospy.logerr('Could not lookup a parameter for track roomba task')
             raise
 
+        if self._DEBUG:
+            self._debug_pub = rospy.Publisher('~debug_height_hold', Vector3Stamped, queue_size=10)
+
         if self._DESIRED_HEIGHT is not None and self._DESIRED_HEIGHT < self._MIN_MANEUVER_HEIGHT:
             raise ValueError('Desired height was below the minimum maneuver height')
+        self._deadzone_activated = False
 
     # uses a p-controller to return a velocity to maintain a height
     def get_height_hold_response(self, height):
         with self._lock:
             if self._DESIRED_HEIGHT is None:
                 raise ValueError('No height to hold')
             self._delta_z = self._DESIRED_HEIGHT - height
-            return self._K_Z * self._delta_z
+
+            if self._DEBUG:
+                msg = Vector3Stamped()
+                msg.header.stamp = rospy.Time.now()
+                msg.vector.x = self._DESIRED_HEIGHT
+                msg.vector.y = height
+                msg.vector.z = self._delta_z
+                self._debug_pub.publish(msg)
+
+            if abs(self._delta_z) < self._DEADZONE:
```

Comment:
what is `deadzone` ? A comment on what it is would be helpful.

Reply:
Fixed